packages/@uppy/dashboard/src/components/VirtualList.js

```diff
@@ -102,13 +88,32 @@ class VirtualList extends Component {
     this.resize()
   }
 
-  componentDidMount () {
+  componentWillUnmount () {
+    window.removeEventListener('resize', this.handleResize)
+  }
+
+  handleScroll = () => {
+    const { sync } = this.props
+
+    this.setState({ offset: this.base.scrollTop })
+
+    if (sync) {
+      this.forceUpdate()
+    }
```

Comment:
I don't think this would still work, as `setState` is now async. This would force the render of the previous state I think. Instead, we should be using `shouldComponentUpdate()` – or maybe drop this option altogether.

Reply:
btw I didn't change anything here besides the order of the functions because eslint is complaining. But we can check if it still works by using the instagram plugin.

Reply:
@aduh95 - correct, this will mark the next render as "forced" and bypass shouldComponentUpdate. My guess is the previous `sync` option was leveraging the fact that `forceUpdate()` in Preact 8 forced a synchronous render pass, which was removed for performance reasons. Definitely seems worth considering removing that option.